service/history/timer_queue_active_task_executor.go

```diff
@@ -310,6 +277,66 @@ Loop:
 	return t.updateWorkflowExecution(ctx, weContext, mutableState, scheduleWorkflowTask)
 }
 
+func (t *timerQueueActiveTaskExecutor) processTimeoutTask(
+	mutableState workflow.MutableState,
+	timerSequenceID workflow.TimerSequenceID,
+	ai *persistencespb.ActivityInfo,
+	updateMutableState *bool,
+	scheduleWorkflowTask *bool,
```

Comment:
I think these should be return values.

Reply:
sigh. it is ugly both ways to me...

Reply:
actually - I tried - return values are even more ugly. 

it will be either named return values (which is , to my best knowledge, something go community  currently recommend not to do), or plain `return true, false, nil`, with is error prone and not readable.

Best thing will be to extract all of that to a separate structure. It started good....but then we have a random emitTimeoutMetricScopeWithNamespaceTag call. 
Which cannot be extracted (used in few other places), cannot be converted into separate function, and also need to be wrapped into some sort of interface because "testing"

Reply:
Agree this should be extracted to return values. You can create a struct to make it more readable.

Reply:
ok :)
still ugly though...